File: LeetCode/2779/solution.cpp
// 2779. Maximum Beauty of an Array After Applying Operation
class Solution {
public:
    int maximumBeauty(vector<int>& nums, int k) {
        int n = nums.size();
        int ans = 0;

        sort(nums.begin(), nums.end());

        for(int i=0, j=0; i<n; i++) {
            while(j<n && nums[j]-nums[i] <= 2*k) {
                j++;
            }
            ans = max(ans, j-i);
        }

        return ans;
    }
};

File: LeetCode/2779/solution.py
# 2779. Maximum Beauty of an Array After Applying Operation
class Solution:
    def maximumBeauty(self, nums: List[int], k: int) -> int:
        nums.sort()

        n = len(nums)
        j, ans = 0, 0

        for i in range(n):
            while j<n and nums[j]-nums[i] <= 2*k:
                j += 1
            ans = max(ans, j-i)

        return ans